test.js

```diff
@@ -104,21 +105,36 @@ describe('dat-gateway', function () {
   })
 
   it('should open a websocket to /peers', function () {
-    const req = http.get('http://localhost:5917/bunsen.hashbase.io/')
-    req.on('error', console.log)
-    const wsUrl = `ws://localhost:3000/peers`
-    try {
-      const socket = websocket(wsUrl, null, null)
-      socket.on('data', function (rawMsg) {
-        var str = String.fromCharCode.apply(null, rawMsg)
-        let msgArray = str.split(':')
-        let count = msgArray[msgArray.length - 1]
-        assert.ok(count > 0, 'count must be non-zero')
-        socket.destroy()
-      })
-    } catch (e) {
-      console.error(e)
+    let socket = null
+    return new Promise((resolve, reject) => {
+      // first populate repo with a dat.
+      axios.get('http://localhost:5917/bunsen.hashbase.io/')
+        .then(res => {
+          setTimeout(() => {
```

Comment:
Since you're including libraries anyway, how about using the [delay](https://www.npmjs.com/package/delay) module to create a promise that does the setTimeout behavior for you?

Reply:
I did some backtracking - not sure even if the setTimeout is needed. My latest commit removes it. Weird behaviour with console.log - quite often when they are commented out, the test fails. But that has happened both with and without the setTimeout.

Reply:
Weird. The `get` should already be adding a 3 second delay from dat-gateway that is there to wait for peers to be discovered.